### PR TITLE
NEW possibility to invert the result of ajax set/del constant function

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

**Comment by hregis** [comment]
@eldy 
this is mandatory for multicompany 8, thanks

**Comment by eldy** [comment]
No change on framework or internal function possible in v8 if it is not a bug fix.

Also content of lib_head.js.php is a jabascript lib that is cached. We must avoid to but dynamic content like translation into it. So label of yes / no button is better defined into the ajax.lib instead of js code (like it is currently) because we must try to have a js code that is the most possible static and not dependent of context.

Also why having a function to manage off/on instead of on/off. What don't you just invert the logic in the code that manipulate constant ?

**Comment by hregis** [comment]
@eldy 
I restored a piece of code, I left the method "hideshow" that allows to do the opposite, very useful in some situations, and the ability to customize the size of the combobox. There is nothing that could break the operation.

**Comment by hregis** [comment]
@eldy 
I want to make an element visible or invisible by activating an "on/off" button, sometimes it will have to be visible / invisible when it is "on" and vice versa with "off".
the current function does not make an element invisble when using the "set" and make a visible element when using the "del".

**Comment by eldy** [comment]
When we set a value, the value is inserted into database and when we unset, we delete.
It is what we want.
Why making invisible when you set ?
If you need this, just called your constant 
MYMODULE_HIDE_ELEMENT
So when you set the constant to "hide", it is set to 1 (option is on)
And in your code, test of   if ! MYMODULE_HIDE_ELEMENT then it is visible.

There is no need to duplicate code

**Comment by hregis** [comment]
@eldy 
je ne vois pas en quoi c'est gênant de pouvoir faire l'inverse, surtout que c'est "showhide" et "hideshow" !! pas de confusion possible ! autant faire simple que de commencer à mettre des constantes et les vérifier !

**Comment by hregis** [comment]
@eldy 
surtout que c'est un fonctionnement à la volée, pas de rechargement de page, donc pas de chargement de nouvelles constantes !
je ne vois pas en quoi ça serait gênant de pouvoir faire l'inverse ?
